Sources_v3/StreamChat/Models/Attachment.swift

```diff
@@ -0,0 +1,301 @@
+    /// A title.
+    public let title: String
+    /// An author.
+    public let author: String?
+    /// A description text.
+    public let text: String?
+    /// A type (see `AttachmentType`).
+    public let type: AttachmentType
+    /// Actions from a command (see `Action`, `Command`).
+    public let actions: [AttachmentAction]
+    /// A URL.
+    public let url: URL?
+    /// An image preview URL.
+    public let imageURL: URL?
+    /// A file description (see `AttachmentFile`).
+    public let file: AttachmentFile?
+    /// An extra data for the attachment.
+    public let extraData: ExtraData
+    
+    public init(
+        title: String,
+        author: String?,
+        text: String?,
+        type: AttachmentType,
+        actions: [AttachmentAction],
+        url: URL?,
+        imageURL: URL?,
+        file: AttachmentFile?,
+        extraData: ExtraData
+    ) {
+        self.title = title
+        self.author = author
+        self.text = text
+        self.type = type
+        self.actions = actions
+        self.url = url
+        self.imageURL = imageURL
+        self.file = file
+        self.extraData = extraData
+    }
+        
+    var hash: String {
+        [title, author, text, type.rawValue, url?.absoluteString, imageURL?.absoluteString]
+            .compactMap { $0 }
+            .reduce("", +)
+    }
+}
+
+/// An attachment action, e.g. send, shuffle.
+public struct AttachmentAction: Codable, Hashable {
+    /// A name.
+    public let name: String
+    /// A value of an action.
+    public let value: String
+    /// A style, e.g. primary button.
+    public let style: ActionStyle
+    /// A type, e.g. button.
+    public let type: ActionType
+    /// A text.
+    public let text: String
+    
+    /// Init an attachment action.
+    /// - Parameters:
+    ///   - name: a name.
+    ///   - value: a value.
+    ///   - style: a style.
+    ///   - type: a type.
+    ///   - text: a text.
+    public init(
+        name: String,
+        value: String,
+        style: ActionStyle,
+        type: ActionType,
+        text: String
+    ) {
+        self.name = name
+        self.value = value
+        self.style = style
+        self.type = type
+        self.text = text
+    }
+    
+    /// An attachment action type, e.g. button.
+    public enum ActionType: String, Codable {
+        case button
+    }
+
+    /// An attachment action style, e.g. primary button.
+    public enum ActionStyle: String, Codable {
+        case `default`
+        case primary
+    }
+}
+
+/// An attachment type.
+/// There are some predefined types on backend but any type can be introduced and sent to backend.
+public enum AttachmentType: RawRepresentable, Codable, Hashable, ExpressibleByStringLiteral {
+    /// Backend specified types.
+    case image
+    case imgur
+    case giphy
+    case video
+    case audio
+    
+    /// Custom types.
+    case youtube
+    case product
+    case file
+    case link
```

Comment:
So, we should sync with other SDKs to make sure they use the same types?

Reply:
Yep, we need to align this at some point.